esmvalcore/cmor/_fixes/fix.py

```diff
@@ -7,23 +7,20 @@
 
 
 class Fix:
-    """
-    Base class for dataset fixes.
-    """
-    def __init__(self, vardef):
+    """Base class for dataset fixes."""
+    def __init__(self, vardef, **extra_facets):
```

Comment:
This modification it's also not needed if you do what is suggested by @bouweandela

Reply:
Sorry, I still don't see that. The change suggested by Bouwe simplifies lines 276­-286 somewhat but has no impact before or after that, right?

Reply:
Sorry, I got mixed: vardef is straight from cmor, so you must left it untouched. 

What I am suggesting here is that Instead of passing the extra_facets, just pass the full variable that you have in the recipe and that now you will be passing to the fix, so we don't only have what is coming from the dict but also potential changes made at recipe level

Reply:
If I understand you correctly, then this amounts to do the change we are discussing above and possibly changing the name of the kwargs here, correct?

Reply:
The full variable is now being passed. Please comment again if this is still not sufficient in your eyes.